configs/components/facter.rb

```diff
@@ -155,19 +155,20 @@
     cmake = "/opt/pl-build-tools/i386-pc-solaris2.#{platform.os_version}/bin/cmake"
 
     # FACT-1156: If we build with -O3, solaris segfaults due to something in std::vector
-    special_flags = "-DCMAKE_CXX_FLAGS_RELEASE='-O2 -DNDEBUG'"
+    special_flags += " -DCMAKE_CXX_FLAGS_RELEASE='-O2 -DNDEBUG' "
```

Comment:
Not really a fan of having to pre- and append whitespace to these.

Also notice we don't do it above in the osx conditional. Should be consistent at least. We can work on concatenating with spaces later I think/.